File: app/src/main/java/karrel/com/bluetoothmanager/presenter/MainPresenter.java
package karrel.com.bluetoothmanager.presenter;

import android.bluetooth.BluetoothDevice;

import org.jetbrains.annotations.Nullable;

/**
 * Created by Rell on 2018. 3. 23..
 */

public interface MainPresenter {
    void searchBluetoothDevices();

    void enabledBluetooth();

    void dissableBluetooth();

    void connectBluetooth(BluetoothDevice data);

    void disconnectBluetooth();

    interface View {

        void requireEnableBt();

        void clearLog();

        void addSearchedDevice(@Nullable BluetoothDevice device);
    }
}

File: app/src/main/java/karrel/com/bluetoothmanager/presenter/MainPresenter.kt
package karrel.com.bluetoothmanager.presenter

import android.bluetooth.BluetoothDevice

/**
 * Created by Rell on 2018. 3. 23..
 */

interface MainPresenter {
    fun searchBluetoothDevices()

    fun enabledBluetooth()

    fun dissableBluetooth()

    fun connectBluetooth(data: BluetoothDevice)

    fun disconnectBluetooth()

    fun onResume()
    fun onStop()
    fun onDestroy()

    interface View {

        fun requireEnableBt()

        fun clearLog()

        fun addSearchedDevice(device: BluetoothDevice?)
    }
}
